subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformLoggingIntegrationTest.groovy

```diff
@@ -98,4 +100,33 @@ class JUnitPlatformLoggingIntegrationTest extends JUnitPlatformIntegrationSpec
         outputContains("TopLevelClass > NestedClass > nestedTestMethod()")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/5975")
+    def "should log display names for dynamically created tests"() {
+        given:
+        file("src/test/java/org/gradle/JUnitJupiterDynamicTest.java") << """
+            package org.gradle;
+            import org.junit.jupiter.api.DynamicTest;
+            import org.junit.jupiter.api.TestFactory;
+            import java.util.stream.IntStream;
+            import java.util.stream.Stream;
+            import static org.junit.jupiter.api.Assertions.*;
+            import static org.junit.jupiter.api.DynamicTest.dynamicTest;
+            public class JUnitJupiterDynamicTest {
+                @TestFactory
+                Stream<DynamicTest> streamOfTests() {
+                    return IntStream.of(2, 4, 5)
+                        .mapToObj(v -> dynamicTest(v + " is even", () -> assertEquals(0, v % 2)));
+                }
+            }
+        """
+
+        when:
+        runAndFail("test")
+
+        then:
+        def prefix = "JUnitJupiterDynamicTest > streamOfTests() > org.gradle.JUnitJupiterDynamicTest"
```

Comment:
❓ Should this really include the test class name?

```suggestion
        def prefix = "JUnitJupiterDynamicTest > streamOfTests() > "
```

Reply:
You're right, it shouldn't. I'll look into why the display name is including the class name in this case.

Reply:
Fixed 👍